app/services/record_declarations/ecf.rb

```diff
@@ -5,19 +5,24 @@ module ECF
     extend ActiveSupport::Concern
 
     included do
-      delegate :participant?, to: :user, allow_nil: true
+      extend ECFClassMethods
     end
 
-    def declaration_model
-      ParticipantDeclaration::ECF
-    end
+    module ECFClassMethods
+      def declaration_model
+        ParticipantDeclaration::ECF
+      end
 
-    def actual_lead_provider
-      user_profile&.school_cohort&.lead_provider&.cpd_lead_provider
+      def valid_declaration_types
+        %w[started completed retained-1 retained-2 retained-3 retained-4]
+      end
     end
 
-    def valid_declaration_types
-      %w[started completed retained-1 retained-2 retained-3 retained-4]
+    def valid_courses
+      [
+        ("ecf-induction" if user&.early_career_teacher?),
+        ("ecf-mentor" if user&.mentor?),
+      ].flatten
```

Comment:
This is a great example of why I am not a fan of mixing class and instance level methods.

There is a concept of class level `valid_courses` - at least on NPQ side. Then there is a concept of instance-level `valid_courses` - right here.

And then there is an unholy mix of the two (in NPQ), to make some other place which expects instance-level method work,
```
    def valid_courses
      self.class.valid_courses
    end
```

I think the solution here is to pick one level and stick with it through thick and thin. And in the wider picture, instance-level methods and validation is the right tool for the job. So the question is: is it worth introducing the class level methods at all? 

Sure, they indicate that some data is static, but if we end up funneling them to instance level methods, then are they worth the effort? For what it is worth, remembering whether you are dealing with class or instance level is an added cost when trying to understand the code.